server/stream.go

```diff
@@ -132,12 +161,23 @@ type stream struct {
 	lqsent  time.Time
 }
 
+type sourceInfo struct {
+	name string
+	sub  *subscription
+	sseq uint64
+	dseq uint64
+	lag  uint64
+	err  *ApiError
+	last time.Time
+}
+
 // Headers for published messages.
 const (
 	JSMsgId             = "Nats-Msg-Id"
 	JSExpectedStream    = "Nats-Expected-Stream"
 	JSExpectedLastSeq   = "Nats-Expected-Last-Sequence"
 	JSExpectedLastMsgId = "Nats-Expected-Last-Msg-Id"
+	JSStreamSource      = "Nats-Stream-Source"
```

Comment:
I saw these and wasnt sure what the user story this fixes? Would the user parse this subjet (its the ack subject) to figure out metadata? This would have original time stamp I guess, the mirrored timestamp would be different?

Reply:
They could, we need it for sources to properly figure out where to pick back up for any given source. We walk backwards through our own stream and look at that header for stream sequence for the source stream. Then we recreate the consumer with that starting sequence +1.

Reply:
OK, its like 70 bytes overhead per message. We could pick out just the bits we need and put it in a header we dont show the user - and so with a much shorter name?

Usually I am not too concerned about messages sizes - a lot less than you :P - but 70/message is a lot.

Reply:
```+----------------+---------+----------+-----------+----------+---------+------+---------+----------------------+
| Stream         | Storage | Template | Consumers | Messages | Bytes   | Lost | Deleted | Cluster              |
+----------------+---------+----------+-----------+----------+---------+------+---------+----------------------+
| ORDERS_6       | File    |          | 5         | 0        | 0 B     | 0    | 0       | n1-c1, n2-c1, n3-c1* |
| ORDERS_7       | File    |          | 5         | 0        | 0 B     | 0    | 0       | n1-c1, n2-c1*, n3-c1 |
| ORDERS_2       | File    |          | 5         | 0        | 0 B     | 0    | 0       | n1-c1, n2-c1*, n3-c1 |
| ORDERS_3       | File    |          | 5         | 0        | 0 B     | 0    | 0       | n1-c1, n2-c1*, n3-c1 |
| ORDERS_4       | File    |          | 5         | 0        | 0 B     | 0    | 0       | n1-c1*, n2-c1, n3-c1 |
| ORDERS_5       | File    |          | 5         | 0        | 0 B     | 0    | 0       | n1-c1*, n2-c1, n3-c1 |
| ORDERS_8       | File    |          | 5         | 0        | 0 B     | 0    | 0       | n1-c1, n2-c1*, n3-c1 |
| ORDERS_9       | File    |          | 5         | 0        | 0 B     | 0    | 0       | n1-c1, n2-c1*, n3-c1 |
| ORDERS_0       | File    |          | 6         | 100      | 4.5 KiB | 0    | 0       | n1-c1, n2-c1*, n3-c1 |
| ORDERS_1       | File    |          | 6         | 100      | 4.5 KiB | 0    | 0       | n1-c1*, n2-c1, n3-c1 |
| ORDERS_ARCHIVE | File    |          | 0         | 200      | 27 KiB  | 0    | 0       | n2-c2*               |
+----------------+---------+----------+-----------+----------+---------+------+---------+----------------------+```

Worst case since my messages are tiny though

Reply:
After our discussion let me know if you still want me to look into shrinking or do we think this could be valuable as is to end users?

/cc @kozlovic

Reply:
I think its ok, it should be useful. 1 billion messages = 70GB of this header :P

Reply:
Compression ;)